### DISCUSSION TITLE
Junk Dimensions with Partitions

### DISCUSSION THREAD

**Discussion by TinyTimothy2** [TabularEditor/TabularEditor3#1252] [Data Modeling]
Since i have 1 B Fact and 20 M dim, I have  had to resort to junk dimension.
I have partitions by years on Fact table
The great thing about having the data in a regular dimension is you can rely on the data in the dimension to be up to date and don't need to process older partitions. 
However with the junk dimension, you have to apply the surrogate key back to the fact table.
This seems that i would need to update all partitions at least weekly to ensure data is correct vs all years.

Does that seem correct?
Am I perhaps missing an easier way.

Also, it does appear if extend some of the cross joins to create the junk dim, i might exceed 1k rows.
Will that be a problem?

**Comment by otykier** [comment]
Without knowing your specific business model, it’s difficult to say how best to approach this, but it would probably make most sense to keep only attributes that never change in the junk dimension, to ensure that you don’t have to update the surrogate keys back on the fact.

https://www.kimballgroup.com/2009/06/design-tip-113-creating-using-and-maintaining-junk-dimensions/
